src/main/java/frc/bionic/swerve/Vision.java

```diff
@@ -0,0 +1,262 @@
+  }
+
+  /**
+   * Updates network frontCamFeed variables.
+   */
+  private void updateTableVariables() {
+    tv = activeCamFeed.getEntry("tv").getDouble(0.0);
+    tx = activeCamFeed.getEntry("tx").getDouble(0.0);
+    ty = activeCamFeed.getEntry("ty").getDouble(0.0);
+    ta = activeCamFeed.getEntry("ta").getDouble(0.0);
+    ts = activeCamFeed.getEntry("ts").getDouble(0.0);
+    tl = activeCamFeed.getEntry("tl").getDouble(0.0);
+  }
+
+  /**
+   * Sets the led states
+  */
+  public void setLights(final double ledState){
+    this.ledState = ledState;
+  }
+
+  /**
+   * 
+   * Establishes a pipeline based on what the user sees
+  */
+  public void setPipeline(final double pipeline){
+    this.pipeline = pipeline;
+  }
+
+  /**
+   * Gets the X offset in degrees.
+   *
+   * @return The X offset in degrees.
+   */
+  public double getXOffset() {
+    return tx;
+  }
+
+  /**
+   * Gets the Y offset in radians.
+   *
+   * @return The Y offset in radians.
+   */
+  public double getYOffset() {
+    return ty;
+  }
+
+  /**
+   * Gets the target area.
+   *
+   * @return The target area.
+   */
+  public double getTargetArea() {
+    return ta;
+  }
+
+  /**
+   * Gets the target skew.
+   *
+   * @return Target Skew.
+   */
+  public double getTargetSkew() {
+    return ts;
+  }
+
+  /**
+   * Gets the X limelight's latency.
+   *
+   * @return Latency
+   */
+  public double getLatency() {
+    return tl;
+  }
+
+  /**
+   * Returns 1 if a target is visible, returns 0 if there's no target visible.
+   *
+   * @return 1 or 0
+   */
+  public double isTargetVisibleDouble() {
+    return tv;
+  }
+
+  /**
+   * Gets the distance of the target using the formula from the limelight's documentation.
+   *
+   * @return The distance to the target using a formula.
+   */
+  public double calculateDistanceFromCameraHeight(final double targetHeight, final double cameraHeight, final double cameraAngle) {
+    final double methodDistance = (targetHeight - cameraHeight) / Math.tan(Math.toRadians(cameraAngle + getYOffset()));
+    return methodDistance;
+  }
+
+  /**
+   * Gets the distance to the target based of off its area using a slope-intercept formula.
+   *
+   * @return The distance based off of the area of the target.
+   */
+  public double calculateDistanceArea() {
+    return slope * ta + yintercept;
+  }
+
+  /**
+   * Sets the camera feed to stream from the front facing camera.
+   */
+  public void setFrontCamFeed() {
+    this.activeCamFeed = this.frontCamFeed;
+    this.usingFrontCam = true;
+    this.frontLight = this.lightOn;
+    this.backLight = this.lightOff;
+    this.cameraAngle = 0;
+  }
+
+  public void flash() {
+    this.frontLight = this.lightFlash;
+    this.backLight = this.lightFlash;
+  }
+
+  public boolean isUsingFrontCam() {
+    return this.usingFrontCam;
+  }
+
+  public boolean targetAcquired(){
+    if(this.tv == 1){
+        return true;
+    }
+    else{
+        return false;
+    }
```

Comment:
Much simpler code could replace the entire body (line 209-214) of this method: `return this.tv == 1;`